client/web/src/components/login-toggle.tsx

```diff
@@ -81,6 +81,36 @@ function LoginPopoverContent({
   auth: AuthResponse
   newSession: () => Promise<void>
 }) {
+  /**
+   * canConnectOverTS indicates whether the current viewer
+   * is able to hit the node's web client that's being served
+   * at http://${node.IP}:5252. If false, this means that the
+   * viewer must connect to the correct tailnet before being
+   * able to sign in.
+   */
+  const [canConnectOverTS, setCanConnectOverTS] = useState<boolean>(false)
+
+  const checkTSConnection = useCallback(() => {
```

Comment:
I think we need an isLoading var or something here that's attached as a dependency to this callback. I was able to queue several dozen hanging connections by moving my mouse in and out of the component.  

We can also use an AbortController to set a shorter timeout, but I think the long timeout is fine as long as we restrict it to a single active connection at a time.

Reply:
Oh the loading var is a good idea! I'll play with it tomorrow morning

Reply:
Oh the loading var is a good idea! I'll play with it tomorrow morning

Reply:
Added loading state, PTAL!